multiqc/modules/picard/ValidateSamFile.py

```diff
@@ -0,0 +1,369 @@
+
+    general_data = dict()
+    for sample in data:
+        general_data[sample] = {key: data[sample][key] for key in ('ERROR_count', 'WARNING_count', 'file_validation_status')}
+
+        if sample not in self.general_stats_data:
+                self.general_stats_data[sample] = dict()
+        self.general_stats_data[sample].update(general_data[sample])
+
+    self.general_stats_addcols(data=general_data, headers=headers)
+
+
+def _get_general_stats_headers():
+    """
+        Returns a header dict for the general stats collumns
+    """
+    headers = OrderedDict()
+
+    headers['file_validation_status'] = {
+        'title': "File-Validation Status",
+        'description': 'pass/warn/fail',
+        'hidden': False,
+    }
+
+    headers['WARNING_count'] = {
+        'title': "# Validation Warnings",
+        'description': 'number of warnings',
+        'scale': 'Set3',
+        'max': 1,
+        'colour': '255,237,160',
+        'format': '{:.0f}',
+        'hidden': True,
+    }
+
+    headers['ERROR_count'] = {
+        'title': "# Validation Errors",
+        'description': 'number of errors',
+        'scale': 'RdYlGn-rev',
+        'max': 1,
+        'colour': '252,146,114',
+        'format': '{:.0f}',
+        'hidden': True,
+    }
+
+    return headers
+
+
+def _generate_overview_note(pass_count, only_warning_count, error_count, total_count):
+    """
+        Generates and returns the HTML note that provides a summary of validation status.
+    """
+
+    note_html = ['<ul class="list-group">']
+
+    if error_count:
+        note_html.append(
+                '''
+                <li class="list-group-item list-group-item-danger" >
+                    <span class="label label-danger">Fail</span> - {} {sample} {has} file-validity <strong>ERRORS</strong> (and possibly warnings).
+                </li>
+                '''.format(error_count, sample='samples' if error_count > 1 else 'sample', has='have' if error_count > 1 else 'has')
+            )
+    if only_warning_count:
+        note_html.append(
+                '''
+                <li class="list-group-item list-group-item-warning">
+                    <span class="label label-warning">Warning</span> - {} {sample} {has} only file-validity <em>WARNINGS</em>.
+                </li>
+                '''.format(only_warning_count, sample='samples' if only_warning_count > 1 else 'sample', has='have' if only_warning_count > 1 else 'has')
+            )
+    if pass_count:
+        note_html.append(
+                '''
+                <li class="list-group-item list-group-item-success">
+                    <span class="label label-success">Pass</span> - {}/{} {sample} passed.
+                </li>
+                '''.format(pass_count, total_count, sample='samples' if total_count > 1 else 'sample')
+            )
+
+    note_html.append('</ul>')
+
+    return "\n".join(note_html)
+
+
+def _generate_detailed_table(data):
```

Comment:
This table is way too wide, 60 columns with the test data. This isn't really usable. Could you pick around 5-10 columns max to show here?

Perhaps better than this table would be to have a bar chart showing the numbers of errors and warnings per sample.. The reason for the report is to highlight patterns and outlier samples, users can always go back to the original files to dig into the details of every category.